### PR TITLE
Fix: Fixed issue where the wrong recent item was being removed

### LINKED ISSUE #10820: Bug: Removing an item from recent list doesn't work with PowerToys locksmith
### Description

If I have PowerToys file locksmith enabled it will open that on the file trying to remove. If I don't have it on then it doesn't remove the selected item but the last one in the list

### Steps To Reproduce

1. Right click an item in a recent list
2. Click remove
3. See last file from list is removed

### Steps to fix

- Remove selected item from list

### Files Version

2.4.8.0

### Windows Version

10.0.22621.963

### Relevant Assets/Logs

[debug.log](https://github.com/files-community/Files/files/10291334/debug.log)

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes part of #10820

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

File: src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml.cs [hunk 1]
```diff
@@ -147,13 +150,16 @@ private async Task UpdateRecentsList(NotifyCollectionChangedEventArgs e)
...
 		/// </summary>
 		/// <param name="recentItem">The recent item to be added</param>
-		private async Task AddItemToRecentListAsync(RecentItem recentItem, int index = -1)
+		private bool AddItemToRecentList(RecentItem recentItem, int index = -1)
 		{
 			if (!recentItemsCollection.Any(x => x.Equals(recentItem)))
 			{
 				recentItemsCollection.Insert(index < 0 ? recentItemsCollection.Count : Math.Min(index, recentItemsCollection.Count), recentItem);
-				await recentItem.LoadRecentItemIcon();
+				_ = recentItem.LoadRecentItemIcon()
```
Comment (gave92) [inline src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml.cs:158]: Optional change, can revert if asked: do not wait for recent item icon to be loaded.

Comment (yaira2) [reply src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml.cs:158]: Looks good

**Comment by hez2010** [comment]
Changes LGTM, but this doesn't fix the original issue. I can still reproduce the issue with the fix in this PR.

**Comment by gave92** [comment]
The issue about powertoys or the issue of not removing the correct item?
If the second, could you share a video?

**Comment by hez2010** [comment]
> The issue about powertoys or the issue about not removing the correct item? If the second, could you share a screen recording?

The issue about powertoys. Removing items from the recent list can still trigger the locksmith open, and the item didn't get  removed in the end.

**Comment by gave92** [comment]
Ah ok here I'm only fixing the issue with powertoys off. I'm changing the description to not close the issue.

**Review by hez2010 — APPROVED** [review]
LGTM.

**Pull request merged**